### ISSUE TITLE
GIN_MODE set in .env file does not take effect

### ISSUE THREAD

**Issue by mrold** [gin-gonic/gin#1837]
- go version:1.12.1
- gin version (or commit ref):latest
- operating system:10.14.2

## Description
I'v add my application configuration to a .env file , and load it by GoDotEnv , gin can get the env var "PORT" but not "GIN_MODE".
## Screenshots
![image](https://user-images.githubusercontent.com/9475864/55464120-fc1fc100-562c-11e9-9980-1b186aa0eb75.png)

![image](https://user-images.githubusercontent.com/9475864/55464153-0c37a080-562d-11e9-9749-35b1ddfa3c07.png)
```go
package main

import (
	"fmt"
	"github.com/gin-gonic/gin"
	"github.com/joho/godotenv"
	"os"
	"log"
)

func init()  {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env file")
	}

}

func main() {
	fmt.Println(os.Getenv("PORT"))
	fmt.Println(os.Getenv("GIN_MODE"))

	//gin.SetMode(gin.ReleaseMode)
	r := gin.Default()
	r.GET("/ping", func(c *gin.Context) {
		c.JSON(200, gin.H{
			"message": "pong",
		})
	})
	r.Run() // listen and serve on 0.0.0.0:8080
}
```

**Comment by dmarkham** [comment]
`gin` also reads the environment  in a `init` function and I'm guessing it's running before yours. The best thing to do if your not going to set the Environment variable outside of go, is to use the `gin.SetMode()` after reading in your config.

**Comment by mrold** [comment]
@dmarkham OK, let me try again, thanks so much.
